### PR TITLE
CIS AWS Add a config provider

### PR DESCRIPTION
**Pull request by uri-weisman**
### Summary of your changes
Add a config provider to address rule 3.5 of the CIS AWS benchmark

### Related Issues
- https://github.com/elastic/csp-security-policies/issues/188 

### Checklist
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary README/documentation (if appropriate)

### CODE REVIEW

File: resources/fetchers/logging_fetcher.go [hunk 1]
```diff
@@ -72,3 +89,18 @@ func (r LoggingResource) GetMetadata() (fetching.ResourceMetadata, error) {
 	}, nil
 }
 func (r LoggingResource) GetElasticCommonData() any { return nil }
+
+func (c ConfigResource) GetMetadata() (fetching.ResourceMetadata, error) {
+	return fetching.ResourceMetadata{
+		ID:      c.GetResourceArn(),
+		Type:    fetching.CloudConfig,
```
Comment (olegsu) [inline resources/fetchers/logging_fetcher.go:96]: Why not add it to the logging type? it is in the same cis section so it make sense some how

File: resources/providers/awslib/configservice/configservice.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+func (c Config) GetResourceArn() string {
+	return fmt.Sprintf("config-service-%s-%s", c.region, c.accountId)
+}
+
+func (c Config) GetResourceName() string {
+	if c.Recorders == nil || len(c.Recorders) == 0 || c.Recorders[0].Name == nil {
+		return ""
+	}
+	return *c.Recorders[0].Name
```
Comment (olegsu) [inline resources/providers/awslib/configservice/configservice.go:80]: Does this idiomatic?

File: resources/utils/testhelper/helper.go [hunk 3]
```diff
@@ -26,3 +26,12 @@ func CollectResources[T any](ch chan T) []T {
 
 	return results
 }
+
+func CreateMockClients[T any](client T, regions []string) map[string]T {
```
Comment (olegsu) [inline resources/utils/testhelper/helper.go:30]: 🔥

**Review by olegsu — CHANGES_REQUESTED** [review]

File: resources/fetchers/logging_fetcher.go (hunk 1 above)
Comment (uri-weisman) [reply resources/fetchers/logging_fetcher.go:96]: IMO, The type field should represent a group of resources and be cloud/benchmark agnostic as much as possible.
`Logging` refers to a benchmark section and should be used only if we had to create a hybrid type such as the `Monitoring` type.

File: resources/providers/awslib/configservice/configservice.go (hunk 2 above)
Comment (uri-weisman) [reply resources/providers/awslib/configservice/configservice.go:80]: Anything specific u would change?

Comment (uri-weisman) [reply resources/providers/awslib/configservice/configservice.go:80]: lets go with configservice-<REGION>

**Review by olegsu — APPROVED** [review]

**Pull request merged**